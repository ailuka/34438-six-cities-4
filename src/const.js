export const EstateType = {
  APARTMENT: `apartment`,
  HOUSE: `house`,
  HOTEL: `hotel`,
  ROOM: `room`,
};

export const CityName = {
  AMSTERDAM: `Amsterdam`,
  BRUSSELS: `Brussels`,
  COLOGNE: `Cologne`,
  DUSSELDORF: `Dusseldorf`,
  HAMBURG: `Hamburg`,
  PARIS: `Paris`,
};

export const MapClass = {
  MAP_MAIN: `cities__map`,
  MAP_DETAILS: `property__map`,
};

export const OfferClassNamesForPageType = {
  main: {
    article: `cities__place-card`,
    image: `cities__image-wrapper`,
    list: `cities__places-list tabs__content`,
    info: ``,
  },
  details: {
    article: `near-places__card`,
    image: `near-places__image-wrapper`,
    list: `near-places__list`,
    info: ``,
  },
  favorites: {
    article: `favorites__card `,
    image: `favorites__image-wrapper`,
    list: ``,
    info: `favorites__card-info`,
  }
};

export const KeyCodes = {
  ENTER: 13,
  SPACEBAR: 32,
};

export const SortTypes = {
  POPULAR: 1,
  PRICE_LOW_TO_HIGH: 2,
  PRICE_HIGH_TO_LOW: 3,
  TOP_RATED_FIRST: 4,
};

export const AppRoute = {
  ROOM: `/offer/:id`,
  LOGIN: `/login`,
  FAVORITE: `/favorite`,
  ROOT: `/`,
};
