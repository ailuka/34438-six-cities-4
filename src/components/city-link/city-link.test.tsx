import * as React from "react";
import * as renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {BrowserRouter as Router} from "react-router-dom";
import {CityName} from "../../const";
import CityLink from "./city-link";

const mockStore = configureStore([]);
const store = mockStore({});

describe(`CityLink snapshot`, () => {
  it(`should render`, () => {
    const tree = renderer.create(
        <Router>
          <Provider store={store}>
            <CityLink
              city={CityName.AMSTERDAM}
              isActive={false}
              isTab={false}
              onCityClick={() => null}
            />
          </Provider>
        </Router>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render with tabs class`, () => {
    const tree = renderer.create(
        <Router>
          <Provider store={store}>
            <CityLink
              city={CityName.AMSTERDAM}
              isActive={true}
              isTab={true}
              onCityClick={() => null}
            />
          </Provider>
        </Router>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
