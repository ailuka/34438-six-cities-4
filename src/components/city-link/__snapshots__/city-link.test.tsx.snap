// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityLink snapshot should render 1`] = `
<a
  className="locations__item-link  "
  href="/"
  onClick={[Function]}
>
  <span>
    Amsterdam
  </span>
</a>
`;

exports[`CityLink snapshot should render with tabs class 1`] = `
<a
  className="locations__item-link tabs__item tabs__item--active"
  href="/"
  onClick={[Function]}
>
  <span>
    Amsterdam
  </span>
</a>
`;
